File: AwaKinG/src/core/Engine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AwaKinG.src.core.engine;

namespace AwaKinG.src.core
{
    public class Engine : IEngine
    {
        List<Entity> Entities = new List<Entity>();
        public Engine()
        {

        }
        public bool Initialize()
        {
            return true;
        }
        public void Shutdown()
        {
            foreach (Entity ent in Entities)
                ent.Shutdown();
        }
        public void Update()
        {
            foreach (Entity ent in Entities)
                ent.Update();
        }
    }
}

File: src/AwaKinG_Engine/src/core/Engine.cpp
#include "Engine.h"

Engine::Engine(){
	_active = false;
}
bool Engine::initialize(){
	_active = true;

	return true;
}
void Engine::shutdown(){

}
void Engine::update(){

}
bool Engine::getActive(){
	return _active;
}
void Engine::setActive(bool value){
	_active = value;
}
